web_external/Viewer/ImageViewerWidgetWrapper/VauiGeoJSImageViewer.js

```diff
@@ -153,10 +154,26 @@ var VauiGeoJSImageViewer = GeojsImageViewerWidget.extend({
                     for (let key in style) {
                         feature.style(key, style[key]);
                     }
+                    feature.geoOn(geo.event.feature.mouseclick, (evt) => {
+                        this._triggerAnnotationEvent(evt.data);
+                    });
                     this.lastFeatureFrame = feature;
                     this.viewer.draw();
                 }
 
+                map.geoOn(geo.event.mouseclick, () => this._triggerAnnotationEvent());
+
+                this._triggerAnnotationEvent = (annotation) => {
+                    if (annotation) {
+                        this._annotationClicks.push(annotation);
+                    }
+                    clearTimeout(this._annotationEventHandle);
```

Comment:
What about instead checking if the timer is set already?

Reply:
I am not seeing anything wrong with the current implementation.

Reply:
Maybe I don't understand the purpose of this logic? It appears you are setting a delayed event handler in order to collect information from this function being called multiple times. If that is the case, you are potentially clearing and resetting the timer before it can fire, which is extra work.

If that is *not* the case, then I don't understand the purpose of the timer.

Reply:
Yes, what you described is the logic. But what specific extra work you are talking about? and how to avoid it?